maintenance-site/index.html

```diff
@@ -137,7 +157,7 @@ <h1>App en maintenance</h1>
     block.style.display = "block"
 
     const redirectToApp = () => {
-      if (source) {
+      if (source && allowedDomains.indexOf(source) !== -1) {
         window.location.replace(source)
```

Comment:
Si `source` contient une des URLs autorisées, ça passe (alors que ça ne devrait pas). Exemple : `?source=https://evil.com/https://passculture.pro`

On pourrait vérifier que la source commence par un des noms de domaine ? Quelque chose comme ça : 

```suggestion
      for (domain of allowedDomains) {
          if (source && source.startsWith(domain)) {
              window.location.replace(source)
          }
```

(Au passage, on devrait peut-être ajouter des slashes à la fin de chaque élément de `allowedDomains`, sinon la source `https://passculture.promo` passe le test et on serait redirigé. :) Mais il faudra aussi vérifier qu'on redirige bien vers la bonne URL (avec un "/" final) depuis le portail pro et les webapp !)

Reply:
> Si source contient une des URLs autorisées, ça passe (alors que ça ne devrait pas). Exemple : ?source=https://evil.com/https://passculture.pro

`indexOf` recherche l’occurrence exacte dans le tableau, donc `https://evil.com/https://passculture.pro` n'est pas dans `allowedDomains`.